ios_sample_test_app/ios_sample_test_app/ViewController.swift

```diff
@@ -0,0 +1,411 @@
+        // qrCode.showShareSheetWithQRCode(from: self, anchor: nil, universalObject: buo, linkProperties: lp) { error in
+            // Showing a share sheet with the QR code
+        //}
+    }
+    
+    // Changes background color and sends Branch event
+    @objc func changeBackgroundColor(_ sender: UIButton) {
+        let event = BranchEvent.customEvent(withName: "Custom Change Background Color Event")
+        event.eventDescription = "User clicked the puchase button to send a custom purchase event"
+        event.customData = [
+            "Custom_Event_Property_key1" : "Custom_Event_Property_val1",
+            "App_Type" : "Ios_Sample_App"
+        ]
+        event.logEvent()
+        
+        self.showToast(message: "Change Background Color Event sent!", font: UIFont(name: "Raleway-Regular", size: 16)!)
+        view.backgroundColor = colors.randomElement()
+    }
+    
+    private let colors: [UIColor] = [
+        .systemRed,
+        .systemCyan,
+        .systemMint,
+        .systemYellow,
+        .systemGreen,
+        .systemPink,
+        .systemOrange,
+        .systemPurple
+    ]
+    
+    // ---------- Setup Constraints and Navigation Bar ----------
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        NSLayoutConstraint.activate([
+            scrollView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 20),
+            scrollView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: 60),
+            scrollView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 20),
+            scrollView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -20),
+            scrollView.centerXAnchor.constraint(equalTo: self.view.centerXAnchor),
+            
+            contentView.topAnchor.constraint(equalTo: scrollView.topAnchor, constant: 8),
+            contentView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor, constant: -8),
+            contentView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor, constant: 8),
+            contentView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor, constant: -8),
+            contentView.centerXAnchor.constraint(equalTo: scrollView.centerXAnchor),
+            
+            generatedBranchLink.topAnchor.constraint(equalTo: contentView.topAnchor, constant: -8),
+            generatedBranchLink.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 0),
+            generatedBranchLink.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: 0.0),
+            generatedBranchLink.heightAnchor.constraint(equalToConstant: 33),
+
+            branchBadgeImageView.topAnchor.constraint(equalTo: generatedBranchLink.bottomAnchor, constant: 10),
+            branchBadgeImageView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 20),
```

Comment:
```suggestion
                        branchBadgeImageView.centerXAnchor.constraint(equalTo: contentView.centerXAnchor),
```

Since there is a width and leader constraint, the image view is a bit off-centered. Updating the leading constraint to a centerX constraint can fix this,

Reply:
Added center X anchor constraints